main.tf

```diff
@@ -21,11 +21,12 @@ resource "azurerm_virtual_network" "vnet" {
 }
 
 resource "azurerm_subnet" "sb" {
-  count                = "4"
-  name                 = "${var.name}-subnet-${count.index}-${var.env}"
-  resource_group_name  = "${azurerm_virtual_network.vnet.resource_group_name}"
-  virtual_network_name = "${azurerm_virtual_network.vnet.name}"
-  address_prefix       = "${cidrsubnet("${var.source_range}", 4, count.index)}"
+  count                                          = "4"
+  name                                           = "${var.name}-subnet-${count.index}-${var.env}"
+  resource_group_name                            = "${azurerm_virtual_network.vnet.resource_group_name}"
+  virtual_network_name                           = "${azurerm_virtual_network.vnet.name}"
+  address_prefix                                 = "${cidrsubnet("${var.source_range}", 4, count.index)}"
+  enforce_private_link_endpoint_network_policies = var.iaas_subnet_enforce_private_link_endpoint_network_policies
 
   lifecycle {
     ignore_changes = "address_prefix"
```

Comment:
do we need the _lifecycle_ here ?as the PR build erred complaining -  ```
[1m[31mError: [0m[0m[1mInvalid expression[0m 
[0m  on main.tf line 32, in resource "azurerm_subnet" "sb":
   32:     ignore_changes = "[4maddress_prefix[0m"
  [0m
 A static list expression is required.
```

Reply:
I wouldn't have thought so but need to check a plan in all the envs